src/main.rs

```diff
@@ -59,6 +70,112 @@ fn check_valid_env(
+    let exclude = ["PS1", "TERM", "HOME"];
+
+    env::vars()
+        .filter(|&(ref k, _)| !exclude.contains(&k.as_str()))
+        .collect()
+}
+
+fn get_ct_vars(
+    org_id: Option<&str>,
+    env: Option<&str>,
+    environments: &Environments,
+) -> Result<HashMap<String, String>> {
+    // Create a HashMap with all the CloudTruth environment values for this environment.
+    let mut ct_vars = HashMap::new();
+    let parameters = Parameters::new();
+    let env_id = environments.get_id(org_id, env)?;
+    let list = parameters.get_parameter_names(org_id, env_id)?;
+    for key in list.iter() {
+        let parameter = parameters.get_body(org_id, env, key)?;
+        // Put the key/value pair into the environment
+        let value = parameter.unwrap_or_else(|| "".to_string());
+        ct_vars.insert(key.to_string(), value);
+    }
+
+    Ok(ct_vars)
+}
+
+fn process_overrides(overrides: Vec<String>) -> HashMap<String, String> {
+    // Create HashMap with all the user-provided overrides
+    let mut over_vars = HashMap::new();
+    for arg_val in overrides {
+        let temp: Vec<&str> = arg_val.splitn(2, '=').collect();
+        if temp.len() != 2 {
+            println!("Ignoring {} due to  no '='", arg_val);
+            continue;
+        }
+        over_vars.insert(temp[0].to_string(), temp[1].to_string());
+    }
+
+    over_vars
+}
+
+fn process_run_command(
+    org_id: Option<&str>,
+    env: Option<&str>,
+    environments: &Environments,
+    subcmd_args: &ArgMatches,
+) -> Result<()> {
+    let mut sub_proc: Exec;
+    if let Some(command_args) = subcmd_args.subcommand_matches("command") {
+        let command = command_args.value_of("COMMAND").unwrap();
+        sub_proc = Exec::shell(command);
+    } else if let Some(arg_args) = subcmd_args.subcommand_matches("arguments") {
+        let mut arguments = arg_args.values_of_lossy("ARGUMENTS").unwrap();
+        let command = arguments.remove(0);
+        if command.contains(' ') {
+            warn("command contains spaces, and will likely fail.".to_string());
+            let mut reformed = format!("{} {}", command, arguments.join(" "));
+            reformed = reformed.replace("$", "\\$");
+            println!("Try using 'cloudtruth run command \"{}\"'", reformed.trim());
+        }
+        sub_proc = Exec::cmd(command).args(&arguments);
+    } else {
+        warn_missing_subcommand("run");
+        process::exit(0);
+    }
+
+    // setup the environment for the sub-process
+    let preserve = subcmd_args.is_present("preserve");
+    let overrides = subcmd_args.values_of_lossy("set").unwrap_or_default();
+    let removals = subcmd_args.values_of_lossy("remove").unwrap_or_default();
+    let mut env_vars = if !preserve {
+        HashMap::new()
+    } else {
+        current_env()
+    };
+
+    // add breadcrumbs about which environment this came from
+    env_vars.insert("CT_ENV".to_string(), env.unwrap_or("default").to_string());
```

Comment:
You should probably be referencing `config::env::ENV_VAR_PREFIX`. You'll need to adjust the visibility. It might make sense to tuck this into `config::env` so all the code handling the CloudTruth configuration values from the environment is in one centralized place.